### PR TITLE
add oneMKL csr and optimized csr to benchmark

### PR DESCRIPTION
**Pull request by yhmtsai**
This PR adds the oneMKL csr and optimized csr to spmv benchmark.

### CODE REVIEW

File: benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,17 @@ function(ginkgo_benchmark_hipsparse_linops type def)
     target_link_libraries(hipsparse_linops_${type} Ginkgo::ginkgo ${HIPSPARSE_LIBRARIES})
 endfunction()
 
+function(ginkgo_benchmark_onemkl_linops type def)
+    add_library(onemkl_linops_${type} utils/dpcpp_linops.dp.cpp)
+    # make the dependency public to catch issues
+    target_compile_definitions(onemkl_linops_${type} PUBLIC ${def})
+    #find_package(MKL CONFIG REQUIRED HINTS "$ENV{MKLROOT}")
+    target_compile_options(onemkl_linops_${type} PRIVATE $<TARGET_PROPERTY:MKL::MKL_DPCPP,INTERFACE_COMPILE_OPTIONS>)
+    target_include_directories(onemkl_linops_${type} PRIVATE $<TARGET_PROPERTY:MKL::MKL_DPCPP,INTERFACE_INCLUDE_DIRECTORIES>)
```
[the remark this answers is not in the document]
Comment (yhmtsai) [reply benchmark/CMakeLists.txt:55]: It is not automatic, I think. Those are private in the ginkgo_dpcpp

File: benchmark/utils/dpcpp_linops.dp.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            [](oneapi::mkl::sparse::matrix_handle_t mat_handle) {
+                oneapi::mkl::sparse::release_matrix_handle(&mat_handle);
+            });
+    }
+
+    OnemklBase(std::shared_ptr<const gko::Executor> exec,
+               const gko::dim<2>& size = gko::dim<2>{})
+        : gko::LinOp(exec, size)
+    {
+        device_exec_ =
```
[the remark this answers is not in the document]
Comment (yhmtsai) [reply benchmark/utils/dpcpp_linops.dp.cpp:94]: get_device_exec return the device_exec_.
I can always do dynamic_pointer_cast in get_device_exec function, but I do not think it give any difference

File: benchmark/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (upsj) [reply benchmark/CMakeLists.txt:55]: by linking against MKL::MKL_DPCPP, you get all the associated include directories and compile options added automatically, no need to manually extract them.

File: benchmark/utils/dpcpp_linops.dp.cpp [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    friend class gko::EnableCreateMethod<OnemklCsr>;
+    friend class gko::EnablePolymorphicObject<OnemklCsr, OnemklBase>;
+
+public:
+    using Csr = gko::matrix::Csr<ValueType, IndexType>;
+    using mat_data = gko::matrix_data<ValueType, IndexType>;
+    void read(const mat_data& data) override
+    {
+        csr_->read(data);
+        this->set_size(gko::dim<2>{csr_->get_size()});
```
Comment (tcojean) [inline benchmark/utils/dpcpp_linops.dp.cpp:141]: Couldn't you do it without creating a new `gko::dim<2>`?
```suggestion
        this->set_size(csr_->get_size());
```

**Review by tcojean — APPROVED** [review]
LGTM

**Pull request merged**